app.js

```diff
@@ -1,45 +1,73 @@
 const express = require("express");
 const fs = require("fs");
+const bodyParser = require("body-parser")
 
 const app = express();
 
 const port = 5000;
 
-app.get('/', (req, res) => {
-	res.send('Greeting from server');
+let products = [];
+
+fs.readFile('./db/products.json', 'utf-8', (err, data) => {
+	if (err) throw err;
+	products = JSON.parse(data);
 });
 
-app.get('/data', (req, res) => {
-	res.send('Here is some data');
+const writeToFile = (list, res) => {
+	fs.writeFile('./db/products.json', JSON.stringify(list), err => {
+		if (err) throw err;
+		res.send(list);
+	});
+};
+
+app.use(bodyParser.urlencoded({extended: false}));
+
+app.use(bodyParser.json());
+
+app.get('/products', (req, res) => {
+	res.send(products);
 });
 
-app.listen(port, () => {
-	console.log(`Server is listening on port ${port}`);
-})
+app.get('/products/:id', (req, res) => {
+	const { id } = req.params;
 
-let text = '';
-for (let i = 1; i <= 10; i++) {
-	text += i + '\n';
-}
+	const product = products.filter((product) => product.id === id);
```

Comment:
It's not correct to use filter here. Filter returns you array of items that match your condition, which means it will check EVERY element in your array. What if I need to get product with id 1 and there is million products, it will take eternity

Reply:
Yes, it makes sense to use the find method.